cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminService.java

```diff
@@ -154,15 +154,21 @@ public DatasetSpecification call() throws Exception {
     }
   }
 
-  private void writeSystemMetadata(Id.DatasetInstance datasetInstanceId, DatasetSpecification spec,
-                                   DatasetProperties props, DatasetTypeMeta typeMeta, DatasetType type,
-                                   DatasetContext context, boolean existing) throws IOException {
+  private void writeSystemMetadata(Id.DatasetInstance datasetInstanceId, final DatasetSpecification spec,
+                                   DatasetProperties props, final DatasetTypeMeta typeMeta, final DatasetType type,
+                                   final DatasetContext context, boolean existing, UserGroupInformation ugi)
+    throws IOException {
     // add system metadata for user datasets only
     if (isUserDataset(datasetInstanceId)) {
       Dataset dataset = null;
       try {
         try {
-          dataset = type.getDataset(context, spec, DatasetDefinition.NO_ARGUMENTS);
+          dataset = ImpersonationUtils.doAs(ugi, new Callable<Dataset>() {
+            @Override
```

Comment:
Spoke offline; it may be needed to create the dataset instance. Even if its not strictly required, it should be safe, since its not reading/writing any system-owned dataset/file.